### PR TITLE
Account for new location of build_main in Sphinx 1.7

### PR DESCRIPTION
**Pull request by kohr-h**
Needs testing, but at least the old error is gone.

### CODE REVIEW

**Comment by Syntaf** [comment]
Thanks for the PR! I'll get around to it shortly

**Comment by kohr-h** [comment]
Hm, we get the Sphinx error
```
Builder name  html not registered or available through entry point
```
I had that at home as well, so it seems there are more changes in 1.7.

**Comment by Syntaf** [comment]
I'd also like to keep parity with old sphinx version, if possible. I'll try to take a look into this later in the week and see what can be done

**Comment by kohr-h** [comment]
> Hm, we get the Sphinx error
> 
> Builder name  html not registered or available through entry point
> 
> I had that at home as well, so it seems there are more changes in 1.7.

Apparently there were changes inside the `build_main` as well that broke the current way of providing the command line argument for the builder. I pushed a change that fixes the issue, now it seems to work for me with both Sphinx 1.6 and 1.7.

File: travis_sphinx/build.py [hunk 1]
```diff
@@ -33,10 +33,21 @@ def build(ctx, source, nowarn):
...
-    args = ['-b html']
+
+    try:
+        # Location from Sphinx 1.7 on
+        sphinx_build = sphinx.cmd.build.build_main
+        # args have to be specified this way for 1.7+, otherwise one gets
+        # "Builder name  html not registered or available through entry point"
+        args = ['-b', 'html']
+    except AttributeError:
+        # Old location
```
Comment (kohr-h) [inline travis_sphinx/build.py]: Would you prefer an explicit version check?

**Comment by kohr-h** [comment]
Tests use the latest sphinx as pulled in by `pip install travis-sphinx`, maybe a sphinx 1.6 test case should be added to ensure backwards compatibility.

File: travis_sphinx/build.py (hunk 1 above)
Comment (Syntaf) [reply travis_sphinx/build.py]: I think this approach seems fine with me

**Comment by Syntaf** [comment]
Sorry for the late response, i'll be addressing and likely merging this PR tomorrow. Thanks for your work! Looks all good to me, I can go ahead and add another test for 1.6 once this is merged.

**Pull request merged**